src/components/AddUserScreen.js

```diff
@@ -125,104 +143,114 @@ const AddUserScreen = ({navigation, route}: AddUserScreenProps): Node => {
         />
+        <View style={styles.inputContainer}>
+          <Text style={styles.inputLabel}>Gender:</Text>
+          <Picker
+            style={styles.inputValue}
+            itemStyle={styles.pickerItem}
+            selectedValue={user.gender}
+            onValueChange={(val: string | number) =>
+              writeValue(val, 'gender', user)
+            }>
+            <Picker.Item label="male" value="male" />
+            <Picker.Item label="female" value="female" />
+          </Picker>
+        </View>
+        <View style={styles.inputContainer}>
+          <Text style={styles.inputLabel}>Birth date:</Text>
+          <DateTimePicker
+            value={user.dob?.date ? new Date(user.dob.date) : new Date()}
+            onChange={(event, val: string) =>
+              writeValue(new Date(val).toISOString(), 'dob.date', user)
+            }
+            style={styles.inputValue}
+          />
+        </View>
         <UserInput
-          label="State"
-          onChangeText={(val: string) =>
-            writeValue(val, 'location.state', user)
-          }
+          label="Email:"
+          onChangeText={(val: string) => writeValue(val, 'email', user)}
         />
         <UserInput
-          label="Postcode"
-          onChangeText={(val: string) =>
-            writeValue(val, 'location.postcode', user)
-          }
+          label="Phone:"
+          onChangeText={(val: string) => writeValue(val, 'phone', user)}
         />
         <UserInput
-          label="Country"
-          onChangeText={(val: string) =>
-            writeValue(val, 'location.country', user)
-          }
+          label="Cellphone:"
+          onChangeText={(val: string) => writeValue(val, 'cell', user)}
         />
-      </View>
-    </ScrollView>
+        <Text style={styles.locationLabel}>Location:</Text>
+        <View style={styles.locationContainer}>
+          <UserInput
+            label="Number"
+            onChangeText={(val: string) =>
+              writeValue(parseInt(val, 10), 'location.street.number', user)
+            }
+          />
+          <UserInput
+            label="Street"
+            onChangeText={(val: string) =>
+              writeValue(val, 'location.street.name', user)
+            }
+          />
+          <UserInput
+            label="City"
+            onChangeText={(val: string) =>
+              writeValue(val, 'location.city', user)
+            }
+          />
+          <UserInput
+            label="State"
+            onChangeText={(val: string) =>
+              writeValue(val, 'location.state', user)
+            }
+          />
+          <UserInput
+            label="Postcode"
+            onChangeText={(val: string) =>
+              writeValue(val, 'location.postcode', user)
+            }
+          />
+          <UserInput
+            label="Country"
+            onChangeText={(val: string) =>
+              writeValue(val, 'location.country', user)
+            }
```

Comment:
These component are duplicated with EditUserScreen. Please try to separate them into a component

Reply:
And treat above components Gender, Birthday,... as the same

Reply:
Thank you for your feedback. I've refactored and split them into separate reusable components